src/main/java/io/zipcoder/Classroom.java

```diff
@@ -73,15 +151,69 @@ public double getClassAverage(){
      * @return
      */
     public String getClassScores(){
-        return null;
+        if (this.studentsEnrolled == 0) return "No Students";
+        StringBuilder finalOutput = new StringBuilder("Students:\n");
+        finalOutputAssembler(finalOutput);
+        return finalOutput.toString();
+    }
+
+    private void finalOutputAssembler(StringBuilder finalOutput) {
+        for (Student validStudent : this.students){
+            if (validStudent != null){
+                finalOutput.append(validStudent.getFirstName() + " " + validStudent.getLastName() + " -> "
+                        + validStudent.getAverage() + "\n");
+            }
+        }
     }
 
     /**
      * Sorts the Students array from highest average to lowest, and ties are broken alphabetically.
      */
     public void sortStudentsByScore() {
+        //ArrayList<Student> inputArrayList = new ArrayList<Student>();
+        Student[] inputArray = new Student[studentsEnrolled];
+        int firstIndex = 0;
+        for (Student validStudent:this.students) {
+            if (validStudent != null){
+                inputArray[firstIndex] = validStudent;
+                firstIndex++;
+            }
+        }
+        //listSorter(inputArrayList);
+        sortInputArray(inputArray);
+
+    }
+
+    private void sortInputArray(Student[] inputArray) {
+        Arrays.sort(inputArray, studentComparator);
+        int index = 0;
+        for (Student sortedStudent:inputArray) {
+            this.students[index] = sortedStudent;
+            System.out.println(sortedStudent.getFirstName());
+            index++;
+        }
+    }
+
+//    private void listSorter(ArrayList<Student> inputArrayList) {
+//        Collections.sort(inputArrayList, new Comparator<Student>(){
+//            public int compare(Student s1, Student s2) {
+//                return s2.getAverage().compareTo(s1.getAverage());
+//            }
+//        });
+//    }
+
+    private void listSorter(ArrayList<Student> inputArrayList){
 
     }
+    Comparator<Student> studentComparator = new Comparator<Student>() {
+        public int compare(Student o1, Student o2) {
+            double difference = o2.getAverage() - o1.getAverage();
+            if (difference == 0.0){
+                difference = o1.getLastName().compareTo(o2.getLastName());
+            }
+            return ((int) difference);
+        }
+    };
```

Comment:
you should "implements" this at the top of the class. that way the compiler can check your work.